src/package/packageVersionDependency.ts

```diff
@@ -0,0 +1,382 @@
+   */
+  public async getDependencyDotProducer(): Promise<DependencyDotProducer> {
+    const isValid = await this.validatePackageVersion();
+    if (!isValid) {
+      throw messages.createError('invalidPackageVersionIdError', [this.resolvedPackageId]);
+    }
+
+    const query = `${SELECT_PACKAGE_VERSION_DEPENDENCY} WHERE Id = '${this.resolvedPackageId}'`;
+    const result = await this.connection.tooling.query<{
+      CalcTransitiveDependencies: boolean;
+      DependencyGraphJson: string;
+    }>(query);
+
+    const producer = new DependencyDotProducer(
+      this.connection,
+      result.records[0].DependencyGraphJson,
+      this.verbose,
+      this.edgeDirection,
+      this.resolvedPackageId
+    );
+    await producer.init();
+    return producer;
+  }
+
+  /**
+   * Resolves id input to a 08c. User can input a 08c or 04t.
+   * Currently a 04t is not supported in filtering the Package2VersionCreateRequest. So a user's input of 04t will be resolved to a 05i and then a 08c.
+   */
+  private async resolvePackageId(): Promise<void> {
+    const userPackageId = this.project?.getPackageIdFromAlias(this.packageId) ?? this.packageId;
+
+    // User provided a Package2VersionCreateRequest ID (08c) and doesn't need to be resolved
+    if (userPackageId.startsWith('08c')) {
+      this.resolvedPackageId = userPackageId;
+    }
+    // User provided a SubscriberPackageVersionId (04t) and needs to be resolved to a Package2VersionCreateRequest ID (08c)
+    else if (userPackageId.startsWith('04t')) {
+      // First find the Package2Version ID (05i)
+      const query05i = `SELECT Id FROM Package2Version WHERE SubscriberPackageVersionId = '${userPackageId}'`;
+      const result05i = await this.connection.tooling.query<{ Id: string }>(query05i);
+      if (result05i.records?.length !== 1) {
+        throw messages.createError('invalidPackageVersionIdError', [userPackageId]);
+      }
+      const package2VersionId = result05i.records[0].Id;
+
+      // Finally resolve to the Package2VersionCreateRequest ID (08c)
+      const query08c = `SELECT Id FROM Package2VersionCreateRequest WHERE Package2VersionId = '${package2VersionId}'`;
+      const result08c = await this.connection.tooling.query<{ Id: string }>(query08c);
+      if (result08c.records?.length !== 1) {
+        throw messages.createError('invalidPackageVersionIdError', [userPackageId]);
+      }
+      this.resolvedPackageId = result08c.records[0].Id;
+    } else {
+      throw messages.createError('invalidPackageVersionIdError', [userPackageId]);
+    }
+  }
+
+  /**
+   * Checks that the given Package2VersionCreateRequest ID (08c)
+   * 1) exists for the given devhub org
+   * 2) contains the calculateTransitiveDependencies boolean set to true
+   * 3) has a corresponding DependencyGraphJson
+   *
+   * @returns true if DOT code can be generated, false otherwise.
+   */
+  private async validatePackageVersion(): Promise<boolean> {
```

Comment:
Another good place to break up this function into smaller functions, maybe combining the last two into one since they're similar.

E.g., something like:
```
verifyCreateRequestExistsOnDevHub()
verifyTransitiveDependenciesCalculated()
```